src/utils/safeHref.ts

```diff
@@ -0,0 +1,56 @@
+export interface URLInfo {
+    url: string | undefined
+    sameOrigin: boolean
+}
+
+let useSafeHref = false
+
+export function setUseSafeHref(value: boolean) {
+    useSafeHref = value
+}
+
+export function getUseSafeHref(): boolean {
+    return useSafeHref
+}
+
+export function getURLInfo(url: string): URLInfo {
+    /**
+     * An array of the safely allowed url protocols
+     */
+    const safeProtocols = ['http:', 'https:', 'mailto:', 'tel:']
+
+    /**
+     * Create an anchor element to parse the protocol and origin of the url provided
+     */
+    const anchorElement = document.createElement('a')
+    anchorElement.href = url
+    const sameOrigin = anchorElement.origin === window.location.origin
+
+    /**
+     * Verify passed url is using a safe protocol
+     */
+    const isSafeProtocol = safeProtocols.includes(anchorElement.protocol)
```

Comment:
How are protocol-less urls or url fragments handled cross-browser?

Reply:
those will be put into the anchor element which will give it a protocol and an origin, which will be whatever the site uses

Reply:
Let me know if you have further questions about this, would be happy to talk through it with you :)